### PR TITLE
Added board - microS2

### PR DESCRIPTION
**Pull request by microDev1**
Added board specific config files for esp32s2 based microS2.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -410,6 +410,7 @@ jobs:
         board:
         - "espressif_saola_1_wroom"
         - "espressif_saola_1_wrover"
+        - "micro_s2"
```
Comment (tannewt) [inline .github/workflows/build.yml:413]: Please rename the board name to include your brand as well as the board name so `microdev_micro_s2` or something similar. I'm expecting many S2 boards and `micro_s2` is pretty generic on its own.

File: ports/esp32s2/boards/micro_s2/mpconfigboard.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+//Micropython setup
+
+#define MICROPY_HW_BOARD_NAME       "microS2"
```
Comment (dhalbert) [inline ports/esp32s2/boards/micro_s2/mpconfigboard.h]: Include the full name here too (with "microdev", capitalized correctly). For instance, one of our boards is the "Adafruit Trinket M0".

File: .github/workflows/build.yml (hunk 1 above)
Comment (microDev1) [reply .github/workflows/build.yml]: Thanks for your suggestions. I made the necessary changes to include microDev in the board name.

File: ports/esp32s2/boards/micro_s2/mpconfigboard.h (hunk 2 above)
Comment (microDev1) [reply ports/esp32s2/boards/micro_s2/mpconfigboard.h]: Thanks for your suggestions. I made the necessary changes to add microDev to the board name.

**Comment by dhalbert** [comment]
The build was failing due to some trailing whitespace in a few files. I edited most of them in github, but making the `sdkconfig` file empty in github failed (some kind of github bug). So I cloned and pushed the changes.

@microDev1 If you can set your editor to automatically delete trailing whitespace on save, that will prevent these kinds of minor issues. Some editors have a setting, and some have a plugin to do this.

**Comment by microDev1** [comment]
@dhalbert Just added PSRAM support to sdkconfig

**Comment by dhalbert** [comment]
Hi, you need to pull my changes from your repo, since your push put back the trailing whitespace.

**Pull request merged**